validator/db/kv/attestation_history_v2.go

```diff
@@ -170,6 +170,24 @@ func MarkAllAsAttestedSinceLatestWrittenEpoch(
 	return newHD, nil
 }
 
+// AttestedPublicKeys retrieves all public keys in our attestation history bucket.
+func (store *Store) AttestedPublicKeys(ctx context.Context) ([][48]byte, error) {
+	ctx, span := trace.StartSpan(ctx, "Validator.ProposedPublicKeys")
+	defer span.End()
+	var err error
+	attestedPublicKeys := make([][48]byte, 0)
```

Comment:
how performant should this method be? We can pre-allocate the slices given the item count in `newHistoricAttestationsBucket`

Reply:
nvm it has to be within the tx